### PR TITLE
Plural descriptions for user-defined tags

### PR DESCRIPTION
**Pull request by darrell-k**
One for Material: @CDrummond requested plural strings for user-defined roles.

### CODE REVIEW

File: Slim/Web/Settings/Server/Behavior.pm [hunk 1]
```diff
@@ -110,16 +110,24 @@ sub handler {
...
 						id => $userDefinedRoles->{$tag} ? $userDefinedRoles->{$tag}->{id} : $customRoleId++,
 						include => $userDefinedRoles->{$tag}  ? $userDefinedRoles->{$tag}->{include} : 1,
 						albumLink => $userDefinedRoles->{$tag}  ? $userDefinedRoles->{$tag}->{albumLink} : 1,
 					};
 
-					if ( !$userDefinedRoles->{$tag} || $userDefinedRoles->{$tag}->{name} ne $customTags->{$tag}->{name} ) {
-						Slim::Utils::Strings::storeExtraStrings([{
+					if ( !$userDefinedRoles->{$tag} || $userDefinedRoles->{$tag}->{name} ne $customTags->{$tag}->{name}
+									|| $userDefinedRoles->{$tag}->{namePlural} ne $customTags->{$tag}->{namePlural} ) {
+						Slim::Utils::Strings::storeExtraStrings(
```
Comment (michaelherger) [inline Slim/Web/Settings/Server/Behavior.pm:121]: Please indent "correctly":

```perl
						Slim::Utils::Strings::storeExtraStrings([
 							{
 								strings => { EN => $customTags->{$tag}->{name} },
 								token   => $tag,
 							},
 							{
 								strings => { EN => $customTags->{$tag}->{namePlural} },
	 							token   => $tag . "_PLURAL",
 							}
 						]);
 						$changed = 1;
```

This way I know that `$changed` is at the same level as the `Slim::Utils::Strings...` statement etc.

File: Slim/Web/Settings/Server/Behavior.pm [hunk 2]
```diff
@@ -110,16 +110,24 @@ sub handler {
 				if ( $tag ) {
 					$customTags->{$tag} = {
 						name => $paramRef->{$key . '_name'} || $tag,
+						namePlural => $paramRef->{$key . '_namePlural'} || $paramRef->{$key . '_name'} || $tag,
```
Comment (michaelherger) [inline Slim/Web/Settings/Server/Behavior.pm]: Could we fall back to `lcfirst($tag)` instead of the all upper case value?

**Review by michaelherger — APPROVED** [review]

**Comment by CDrummond** [comment]
Thanks, but should these also not be used in the Default skin? As in use plural for browse modes (e.g. 'Vocalists') and use singular when listing tags (or showing album contributors, etc.) (e.g. 'Vocalist: Abc')

**Comment by darrell-k** [comment]
Additional Browse Modes has it's own heading setting in the plugin menu. 
The other places you'll see them in Default Skin is as categories in Artists/Release Types, which should be singular, as you're browsing a single artist, and in albumInfo and trackInfo (top of album tracks listing or track information page): here it would ideally change depending on the number of artists in the the role, ie
Bass Player: Jean Jacques Burnel
but
Vocalist**s**: Dave Greenfield, Hugh Cornwell, Jean Jacques Burnel 

I haven't yet tracked down where the strings are set in this last case - deep in the bowels of XML Browser or the HTML templates, I think.

Let me know if I've missed something.

**Pull request merged**